shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -182,7 +182,37 @@ public InitResult call() {
               try (TraceSection e = TraceSection.scoped("FlutterLoader initTask")) {
                 ResourceExtractor resourceExtractor = initResources(appContext);
 
-                flutterJNI.loadLibrary();
+                try {
+                  flutterJNI.loadLibrary();
+                } catch (UnsatisfiedLinkError unsatisfiedLinkError) {
+                  String couldntFindVersion = "couldn't find \"libflutter.so\"";
+                  String notFoundVersion = "dlopen failed: library \"libflutter.so\" not found";
+
+                  if (unsatisfiedLinkError.toString().contains(couldntFindVersion)
+                      || unsatisfiedLinkError.toString().contains(notFoundVersion)) {
+                    // To gather more information for
+                    // https://github.com/flutter/flutter/issues/144291,
+                    // log the contents of the native libraries directory as well as the
+                    // cpu architecture.
+
+                    String cpuArch = System.getProperty("os.arch");
+                    String[] nativeLibs = new File(flutterApplicationInfo.nativeLibraryDir).list();
```

Comment:
should you filter this list down to just files that end with .so or change the message below on line 208?

Reply:
I'll change the message to `the following files` (I suppose I was assuming only .so files end up there because thats all I saw when I tested it, but that is probably a bad assumption)